python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -270,7 +270,22 @@ def _process_col(col, unit, dayfirst, infer_datetime_format, format):
                 column.datetime._numpy_to_pandas_conversion[unit]
             )
             col = col.binary_operator(binop="mul", rhs=factor)
-        col = col.as_datetime_column(dtype="datetime64[ns]")
+
+        if format is not None:
+            # Converting to int because,
+            # pandas actually creates a datetime column
+            # out of float values and then creates an
+            # int column out of it to parse against `format`.
+            # Instead we directly cast to int and perform
+            # parsing against `format`.
+            col = (
+                col.astype("int")
+                .astype("str")
+                .as_datetime_column(dtype="datetime64[ns]", format=format)
```

Comment:
I understand Pandas has these constraints, but any chance we could infer the datetime unit based on the format? I.E. if the format is `%Y/%m/%d` we could safely use `datetime64[s]` instead of going all the way down to nanoseconds.

Reply:
Actually we can always return `datetime64[s]` in this case and only when we encounter `%f` which is a Microsecond format specifier we can return `datetime64[us]`